BedrockServer.cpp

```diff
@@ -991,11 +991,20 @@ void BedrockServer::worker(int threadId)
                         // Roll back the transaction, it'll get re-run in the sync thread.
                         core.rollback();
 
-                        // We're not handling a writable command anymore.
-                        SINFO("Sending non-parallel command " << command->request.methodLine
-                              << " to sync thread. Sync thread has " << _syncNodeQueuedCommands.size()
-                              << " queued commands.");
-                        _syncNodeQueuedCommands.push(move(command));
+                        if (state == SQLiteNode::LEADING) {
+                            SINFO("Sending " << command->request.methodLine << " to sync thread.");
+                            _syncNodeQueuedCommands.push(move(command));
+                        } else if (state == SQLiteNode::STANDINGDOWN) {
+                            SINFO("Need to process command " << command->request.methodLine << " but STANDINGDOWN, moving to _standDownQueue.");
+                            _standDownQueue.push(move(command));
+                        } else if (!_clusterMessenger.sendToLeader(*command)) {
+                            // TODO: Also handle this better?
+                            SWARN("Couldn't escalate command " << command->request.methodLine << " to leader. We are in state: " << STCPNode::stateName(state));
+                            _commandQueue.push(move(command));
+                        } else {
+                            SINFO("Escalating " << command->request.methodLine << " to leader.");
```

Comment:
I am confused where "escalating to leader" happens?

Reply:
Above in:
```
} else if (!_clusterMessenger.sendToLeader(*command)) {
```

I could probably remove the negation from that check and swap the contents of that block and this block.